### DISCUSSION TITLE
Is it okay to not close mongo connection in Next.js example

### DISCUSSION THREAD

**Discussion by gmjjatin** [vercel/next.js#68154] [Polls]
In other examples like Javascript /Nodejs, the coders have closed the mongo connection in a finally block after utilization in try.
But such a thing is not present in the Next.js example

**Comment by Juneezee** [comment]
I assume the example you are talking about is this: https://www.mongodb.com/docs/drivers/node/current/fundamentals/connection/connect/#atlas-connection-example

In that case, you don't need to `.close()` the client after every `.find()`, `.insertOne()`, etc. in your Next.js app. `.close()` [closes the client and its underlying connections](https://mongodb.github.io/node-mongodb-native/6.8/classes/MongoClient.html#close), the next time you want to e.g. `.find()` again, you will need to reconnect to MongoDB. Reusing connections is much more cheaper than reconnecting.

You can see more discussion at this StackOverflow post https://stackoverflow.com/a/14607887/7902371

**Reply by gmjjatin** [reply]
the example that I ended upon from mongo docs is https://github.com/vercel/next.js/blob/canary/examples/with-mongodb/pages/index.tsx
where connection was not closed anywhere.
But yeah your answer helps
